glmtools/glm4d.py

```diff
@@ -54,4 +68,24 @@ def t_test_3d(c, X, B, sigma_2, df):
         two-tailed probability value for each t statistic.
     """
     # Your code code here
-    return
+
+    B_2d = B.reshape((-1, B.shape[-1]))
+
+    c_b_cov = c.dot(npl.pinv(X.T.dot(X))).dot(c)
+
+    t = c.dot(B_2d.T) / np.sqrt(sigma_2.ravel() * c_b_cov)
+
+    t_dist = scipy.stats.t(df=df)
+
+    p_value = np.zeros(np.prod(B.shape[:-1]))
+
+    # Calculate t values
+    for i in range(len(t)):
```

Comment:
This loop will be very slow compared to doing it all in one shot - which you can often do with numpy and scipy routines.  This is called "vectorization".  Here `t_dist.cdf` will accept a vector of t values, so:

```
p_value[i] = 2 * (1 - t_dist.cdf(np.abs(t)))
```

would work for a vector of t values.   You could use that in your 2D `glm` function and then call that from here, so you don't need to duplicate the code.